src/osd/osd_types.cc

```diff
@@ -5721,21 +5721,25 @@ void OSDSuperblock::encode(ceph::buffer::list &bl) const
   encode(purged_snaps_last, bl);
   encode(last_purged_snaps_scrub, bl);
   encode(cluster_osdmap_trim_lower_bound, bl);
+  encode(maps, bl);
   ENCODE_FINISH(bl);
 }
 
 void OSDSuperblock::decode(ceph::buffer::list::const_iterator &bl)
 {
-  DECODE_START_LEGACY_COMPAT_LEN(10, 5, 5, bl);
+  DECODE_START_LEGACY_COMPAT_LEN(11, 5, 5, bl);
   if (struct_v < 3) {
     string magic;
     decode(magic, bl);
   }
   decode(cluster_fsid, bl);
   decode(whoami, bl);
   decode(current_epoch, bl);
-  decode(oldest_map, bl);
-  decode(newest_map, bl);
+  if (struct_v >= 11) {
```

Comment:
This looks wrong -- these fields exist regardless of version.  Have you confirmed with the ceph-deencoder/object corpus that this works correctly with previous version superblocks?

Reply:
Should be unconditionally

```
    epoch_t oldest_map, newest_map;
    decode(oldest_map, bl);
    decode(newest_map, bl);
```

You need them below to initialize maps if the encoding version < 11.  Ensure that the ceph-object-corpus/ceph-deencoder tests catch this mistake before you fix it.

Reply:
Verified with `test/encoding/readable.sh`:
Before suggested change:
```
$ ../src/test/encoding/readable.sh > read_res.tst

$ tail -n 1 read_res.tst                              
FAILED 96 / 9438 tests.

$ cat read_res.tst | grep failed | grep Super |  wc -l
96
```
```

```
---

After change:
```
$ ../src/test/encoding/readable.sh > read_res2.tst

$ tail -n 1 read_res.tst                              
passed 9534 tests.
```